glean_parser/templates/python_server.jinja2

```diff
@@ -0,0 +1,194 @@
+{# The final Go code is autogenerated, but this template is not. Please file bugs! #}
+"""
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+AUTOGENERATED BY glean_parser v{{ parser_version }}. DO NOT EDIT. DO NOT COMMIT.
+"""
+
+from __future__ import annotations
+from datetime import datetime, timezone
+from typing import Any
+from uuid import uuid4
+import json
+
+GLEAN_EVENT_MOZLOG_TYPE = "glean-server-event"
+
+
+{% for ping, metrics_by_type in pings.items() %}
+class {{ ping|camelize }}ServerEventLogger:
+    def __init__(
+        self, application_id: str, app_display_version: str, channel: str
+    ) -> None:
+        """
+        Create {{ ping|camelize }}ServerEventLogger instance.
+
+        :param str application_id: The application ID.
+        :param str app_display_version: The application display version.
+        :param str channel: The channel.
+        """
+        self._application_id = application_id
+        self._app_display_version = app_display_version
+        self._channel = channel
+
+    def _record(
```

Comment:
Nit: `black` formats this as one line, because there are no metrics. We could special case no metrics, but that may be ugly.

Reply:
I'd leave this as is for the sake of readability of the template. We can revisit if this turns out inconvenient for consumers.